services/cognito/app/resources/user_resource.rb

```diff
@@ -6,10 +6,11 @@ class UserResource < Cognito::ApplicationResource
   filter :primary_identifier
 
   filter :query, apply: lambda { |records, value, _options|
-    filter_fields = %w[primary_identifier first_name last_name email_address]
-                    .map { |field| "#{field} ILIKE '%#{value[0]}%'" }
-                    .join(' OR ')
-
+    query_by_id = "id IN (#{value[0]})"
+    query_by_non_id_attrs = %w[primary_identifier first_name last_name email_address]
+                            .map { |field| "#{field} ILIKE '%#{value[0]}%'" }
+                            .join(' OR ')
+    filter_fields = value[0].scan(/\D/).empty? ? query_by_id : query_by_non_id_attrs
```

Comment:
Instead of `scan().empty`, we should use `match?`

Reply:
@rpbaltazar Updated the regex checking.